Add flink logs (#5697)
DataDog/integrations-core: flink/README.md

    metrics.scope.operator: flink.operator
    ```

>      **Note**: The system scopes must be remapped for your Flink metrics to be supported, otherwise they will be submitted as custom metrics.

Reviewer: ```suggestion
     **Note**: The system scopes must be remapped for your Flink metrics to be supported, otherwise they are submitted as custom metrics.
```

3. Configure additional [tags][4] in `<FLINK_HOME>/conf/flink-conf.yaml`. Here is an example of custom tags:

    ```yaml
    metrics.reporter.dghttp.tags: <KEY1>:<VALUE1>, <KEY1>:<VALUE2>
    ```

     **Note**: By default, any variables in metric names are sent as tags, so there is no need to add custom tags for `job_id`, `task_id`, etc.

4. Restart Flink to start sending your Flink metrics to the Agent.

#### Log collection

_Available for Agent >6.0_

> 1. Flink uses the `log4j` logger by default. To activate logging to a file and customize the format edit the `log4j.properties`, `log4j-cli.properties`, `log4j-yarn-session.properties`, or `log4j-console.properties files`. See [flink's defaults][13] for default configurations. For example, a `log4j.properties` would contain this configuration by default:

Reviewer: ```suggestion
1. Flink uses the `log4j` logger by default. To activate logging to a file and customize the format edit the `log4j.properties`, `log4j-cli.properties`, `log4j-yarn-session.properties`, or `log4j-console.properties` file. See [Flink's documentation][13] for default configurations. For example, `log4j.properties` contains this configuration by default:
```

   ```conf
   log4j.appender.file=org.apache.log4j.FileAppender
   log4j.appender.file.file=${log.file}
   log4j.appender.file.append=false
   log4j.appender.file.layout=org.apache.log4j.PatternLayout
   log4j.appender.file.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p %-60c %x - %m%n
   ```

2. By default, our integration pipeline supports the following conversion pattern:

    ```text
    %d{yyyy-MM-dd HH:mm:ss,SSS} %-5p %-60c %x - %m%n
    ```

     An example of a valid timestamp is: `2020-02-03 18:43:12,251`.

     Clone and edit the [integration pipeline][11] if you have a different format.

3. Collecting logs is disabled by default in the Datadog Agent, enable it in your `datadog.yaml` file:

   ```yaml
   logs_enabled: true
   ```

> 4. Add the following configuration block to your `flink.d/conf.yaml` file. Change the `path` and `service` parameter values based on your environment. See the [sample flink.d/conf.yaml][12] for all available configuration options.

Reviewer: ```suggestion
4. Uncomment and edit the logs configuration block in your `flink.d/conf.yaml` file. Change the `path` and `service` parameter values based on your environment. See the [sample flink.d/conf.yaml][12] for all available configuration options.
```

See [metadata.csv][8] for a list of metrics provided by this integration.

##### Log collection

_Available for Agent >6.0_

1. Flink uses the `log4j` logger by default. To activate logging to a file and customize the format edit the `log4j.properties`, `log4j-cli.properties`, `log4j-yarn-session.properties`, or `log4j-console.properties files`. See [flink's defaults][13] for default configurations. For example, a `log4j.properties` would contain this configuration by default:

    ```
      log4j.appender.file=org.apache.log4j.FileAppender
      log4j.appender.file.file=${log.file}
      log4j.appender.file.append=false
      log4j.appender.file.layout=org.apache.log4j.PatternLayout
      log4j.appender.file.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p %-60c %x - %m%n
    ```

2. By default, our integration pipeline supports the following conversion pattern:

    ```
      %d{yyyy-MM-dd HH:mm:ss,SSS} %-5p %-60c %x - %m%n
    ```

>     Clone and edit the [integration pipeline][11] if you have a different format.

Reviewer 2: Do we usually have customers use their own pipeline? Since we encourage the timestamp in the `log4j.properties` configuration above and it's also flink's default timestamp format, maybe we don't need this?

Author: I'm not sure- I included this because I saw it in a different integration that uses log4j and allows users to customize log formats

Reviewer 2: Cool, good to know! Can we also include an example timestamp, I think they might be easier to read than the grok syntax.

Author: Sure!
